src/MatrixClientPeg.js

```diff
@@ -96,13 +104,13 @@ class MatrixClientPeg {
     }
 
     getCredentials() {
-        return [
-            this.matrixClient.baseUrl,
-            this.matrixClient.idBaseUrl,
-            this.matrixClient.credentials.userId,
-            this.matrixClient.getAccessToken(),
-            this.matrixClient.isGuest(),
-        ];
+        return {
```

Comment:
I'd quite like to see some sort of `MatrixClientCredentials` abstraction being defined, because we use it in quite a few places.